chore(showcase): `eslint` & `prettier`

aklkv opened the pull request:
### :pushpin: Summary

* add latest eslint and ~~prettier~~ setup
* bump `pnpm`

<!-- If merged, this PR....
This should be a short TL;DR that includes the purpose of the PR.
-->

### :hammer_and_wrench: Detailed description

<!-- If more details are appropriate, add them here. What code changed, and why? -->

### :camera_flash: Screenshots

<!-- Screenshots always help, especially if this PR will change what renders to the browser -->

### :link: External links

<!-- Issues, RFC, etc. -->
Jira ticket: [HDS-XXX](https://hashicorp.atlassian.net/browse/HDS-XXX)
Figma file: [if it applies]

***

### 👀 Component checklist

- [ ] Percy was checked for any visual regression
- [ ] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed (see [templates here](https://hashicorp.atlassian.net/wiki/spaces/HDS/pages/3243114706/Changelog+authoring+best+practices#Templates))

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

didoo commented:
@aklkv FYI @alex-ju is OO until May 5 so unless it's urgent/blocker I would wait for him to come back (he has much more context about these changes and the previous PRs)

aklkv commented:
@alex-ju is there anything I can help here to move it forward?

alex-ju commented:
> I **strongly** think we need to find a way to isolate the changes to the CSS files and have another tool to do the linting for them. Some of the changes I see are too drastic/draconian, and are not what was collectively decided time ago by the HDS team about CSS linting rules.

The main concern with this PR seems to be around Sass linting. Now, I'm not sure what alternatives we have here.
@didoo would it work for you if @aklkv disables prettier for Sass – keeping the code as is – and we plan to look for alternatives at a later point? or do you prefer having it resolved as part of this PR?

didoo commented:
> @didoo would it work for you if @aklkv disables prettier for Sass

yes, that would be the pragmatic thing to do, for me

aklkv commented:
@didoo @alex-ju I have removed all changes to `*.scss` and added it to prettierignore pattern. Let me know if you would like me to address anything else 🙏

didoo commented:
> @didoo @alex-ju I have removed all changes to `*.scss` and added it to prettierignore pattern. Let me know if you would like me to address anything else 🙏

I'll let @alex-ju have a look (he has more context/knowledge)

aklkv commented:
@alex-ju I had to rebase, can I have another ✅ please
